### PR TITLE
Avoid greenthreads hanging endlessly on "slowloris" attack

### PR DESCRIPTION
**Pull request by dirkmueller**
See https://bugs.launchpad.net/nova/+bug/1361360 for the full
story.

### CODE REVIEW

File: chef/cookbooks/heat/templates/default/heat.conf.erb [hunk 1]
```diff
@@ -910,6 +910,7 @@ connection=<%= @database_connection %>
 
 # If False, closes the client socket connection explicitly. (boolean value)
 #wsgi_keep_alive = true
+wsgi_keep_alive = False
```
Comment (mjura) [inline chef/cookbooks/heat/templates/default/heat.conf.erb]: I think this should be written with lowercase. This is OpenStack configuration convention

File: chef/cookbooks/keystone/templates/default/keystone.conf.erb [hunk 2]
```diff
@@ -688,6 +688,7 @@ admin_port = <%= @bind_admin_port %>
 # If set to false, disables keepalives on the server; all connections will be
 # closed after serving one request. (boolean value)
 #wsgi_keep_alive = true
+wsgi_keep_alive = False
```
Comment (mjura) [inline chef/cookbooks/keystone/templates/default/keystone.conf.erb]: I think this should be written with lowercase. This is OpenStack configuration convention

**Comment by aspiers** [comment]
FWIW it's "slowlOris" :-)

**Comment by mjura** [comment]
+1

**Comment by aplanas** [comment]
+1

(info about [slowloris](https://en.wikipedia.org/wiki/Slowloris_%28software%29))

**Pull request merged**